File: app/src/main/java/igrek/todotree/service/preferences/PropertyDefinition.java
package igrek.todotree.service.preferences;


public enum PropertyDefinition {
	
	lockDB(false);
	
	
	private Object defaultValue;
	private PropertyType type;
	
	PropertyDefinition(String defaultValue) {
		this(PropertyType.STRING, defaultValue);
	}
	
	PropertyDefinition(Boolean defaultValue) {
		this(PropertyType.BOOLEAN, defaultValue);
	}
	
	PropertyDefinition(Integer defaultValue) {
		this(PropertyType.INTEGER, defaultValue);
	}
	
	PropertyDefinition(PropertyType type, Object defaultValue) {
		this.type = type;
		this.defaultValue = defaultValue;
	}
	
	public Object getDefaultValue() {
		return defaultValue;
	}
	
	public PropertyType getType() {
		return type;
	}
}

File: app/src/main/java/igrek/todotree/service/preferences/PropertyDefinition.kt
package igrek.todotree.service.preferences

import igrek.todotree.service.preferences.PropertyType

enum class PropertyDefinition(val type: PropertyType, val defaultValue: Any) {
    lockDB(false), userAuthToken("");

    constructor(defaultValue: String) : this(PropertyType.STRING, defaultValue) {}
    constructor(defaultValue: Boolean) : this(PropertyType.BOOLEAN, defaultValue) {}
    constructor(defaultValue: Int) : this(PropertyType.INTEGER, defaultValue) {}
}